### PR TITLE
Add user agent check to frontend

### PR DESCRIPTION
**Pull request by balloob**
This moves the user agent check to the frontend, as that is also where the code lives that it impacts.

The overriding via url logic will remain in the frontend component.

CC @andrey-git

### CODE REVIEW

**Comment by balloob** [comment]
Will work on the home-assistant part if this gets approved.

**Comment by andrey-git** [comment]
What is the benefit of this?

File: public/__init__.py [hunk 1]
```diff
@@ -1,8 +1,34 @@
-"""AUTO-GENERATED. DO NOT MODIFY"""
+"""Frontend for Home Assistant.
+
+AUTO-GENERATED. DO NOT MODIFY.
```
Comment (andrey-git) [inline public/__init__.py]: This is auto appended, so it is ok to modify the version map.

**Comment by balloob** [comment]
@andrey-git benefit would be that we keep frontend related in the frontend repo. So we can keep the map of what is 'latest' and 'es5' in here.

**Review by andrey-git — APPROVED** [review]

**Comment by balloob** [comment]
Backend PR: https://github.com/home-assistant/home-assistant/pull/12786

**Pull request merged**